dataAccessObjects.py

```diff
@@ -133,6 +133,15 @@ def getImgData(self,imgId):
             #TODO raise exception
             return None
 
+class UserPreferDao(DefaultDao):
+    def generateNewId(self):
+        sql ='SELECT pref_id FROM user_prefer ORDER BY pref_set_time DESC LIMIT 1'
+        res = self.db.query(sql)
+        if len(res):
+            return "pref" + "{0:010d}".format(int(res[0][0][4:]) + 1)
```

Comment:
You can use `'pref{0:010d}'.format(int(res[0][0][4:]) + 1)` for better readability